CMakeLists.txt

```diff
@@ -110,9 +112,16 @@ if(NOT HAS_ACCEPTABLE_FORTIFY)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${_FORTIFY_SOURCE_PREFIX}-U_FORTIFY_SOURCE ${_FORTIFY_SOURCE_PREFIX}-D_FORTIFY_SOURCE=1")
 endif()
 
-add_definitions(-Wall -Wextra -pedantic -Wno-unused-parameter
+if(NOT MSVC)
+  add_definitions(-Wall -Wextra -pedantic -Wno-unused-parameter
     -Wstrict-prototypes -std=gnu99)
-if (MINGW)
+else()
+  add_definitions(/W3)
+  add_definitions(-D_CRT_SECURE_NO_WARNINGS)
+  add_definitions(-D_CRT_NONSTDC_NO_DEPRECATE)
```

Comment:
Oh right of course. Just FYI if neovim ever upgrades to a newer version of CMake I believe [add_compile_options](http://www.cmake.org/cmake/help/latest/command/add_compile_options.html) is the way to go for compiler flags.